### DISCUSSION TITLE
希望像屏蔽BT一样屏蔽发送邮件

### DISCUSSION THREAD

**Discussion by crazypeace** [v2fly/v2ray-core#1672] [Ideas]
protocol 能区别 smtp 吗？
希望像屏蔽BT一样屏蔽发送邮件
以免VPS与朋友共用时，朋友的电脑中了木马啥的，滥发邮件，然后被VPS商封停。
---
请不要建议我帮朋友装杀毒软件之类的，朋友如果懂那么一点点电脑，我就让TA自己搭梯子了。

**Comment by dyhkwong** [comment]
SMTP is a server-speaks-first protocol and server-speaks-first protocols won't work with TCP content sniffers enabled (as [the manual](https://www.v2fly.org/config/inbounds.html#sniffingobject) describes), not to mention something like "SMTP sniffer". Maybe you can block SMTP ports instead. BTW you should get rid of these users as they may leak and abuse your servers in unexpected ways.

**Reply by crazypeace** [reply]
Thank you very much!

**Comment by Cwek** [comment]
带TLS的SMTP端口为465，普通SMTP端口为25，如果标准端口的话，阻止这两个端口应该足够。
SMTP有通信协议的模式，理论上可以识别，但开发者愿不愿意弄是一回事，opensource自己解决了。
带TLS的，都带TLS的，就别指望了。
